plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/SQLEditor.java

```diff
@@ -675,6 +675,27 @@ public boolean isActiveTask() {
         return getTotalQueryRunning() > 0;
     }
 
+    @NotNull
+    @Override
+    public SQLDialect getSQLDialect() {
```

Comment:
Also I believe it's not right to return instances that require two-stage initialization only after the first stage from getter